app/Controller/UsersController.php

```diff
@@ -1224,6 +1209,35 @@ public function login()
         }
     }
 
+    private function _postlogin()
+    {
+      $this->User->extralog($this->Auth->user(), "login");
+      $this->User->Behaviors->disable('SysLogLogable.SysLogLogable');
+      $this->User->id = $this->Auth->user('id');
+      $user = $this->User->find('first', array(
+          'conditions' => array(
+              'User.id' => $this->Auth->user('id')
+          ),
+          'recursive' => -1
+      ));
+      $lastUserLogin = $user['User']['last_login'];
+      unset($user['User']['password']);
+      $user['User']['action'] = 'login';
+      $user['User']['last_login'] = $this->Auth->user('current_login');
+      $user['User']['current_login'] = time();
+      $this->User->save($user['User'], true, array('id', 'last_login', 'current_login'));
+      if (empty($this->Auth->authenticate['Form']['passwordHasher']) && !empty($passwordToSave)) {
+          $this->User->saveField('password', $passwordToSave);
+      }
+      $this->User->Behaviors->enable('SysLogLogable.SysLogLogable');
+      if ($lastUserLogin) {
+          $readableDatetime = (new DateTime())->setTimestamp($lastUserLogin)->format('D, d M y H:i:s O'); // RFC822
+          $this->Flash->info(sprintf('Welcome! Last login was on %s', $readableDatetime));
```

Comment:
i18n? or is sprintf covered automatically ?

Reply:
Nice catch. Just replace sprintf with __ and all good :)